pkg/cvo/status.go

```diff
@@ -347,13 +347,16 @@ func (optr *Operator) syncStatus(ctx context.Context, original, config *configv1
 // unavailable AND the general payload task making progress towards its goal. An operator is given 40 minutes since
 // its last update to go ready, or an hour has elapsed since the update began, before the condition is ignored.
 func convertErrorToProgressing(history []configv1.UpdateHistory, now time.Time, status *SyncWorkerStatus) (reason string, message string, ok bool) {
+	uErr, ok := status.Failure.(*payload.UpdateError)
 	if len(history) == 0 || status.Failure == nil || status.Reconciling || status.LastProgress.IsZero() {
+		if ok && uErr.Progressing {
+			return "", "", true
```

Comment:
This removes the "...but the second-level operator has been at it for over 40m and still hasn't leveled".  We want to drop that limit eventually, but I'm not sure if we're ready to drop it quite yet.  And see also the places touched in openshift/cluster-version-operator#464 that would need updating if we do decide to drop the timeout entirely.

Reply:
Are you sure? I thought I added it in the least impactful way possible. The 40m check is next so the existing code would have already entered this `if` and returned before my mod.